primitives/src/identity_id.rs

```diff
@@ -22,27 +23,21 @@ const _POLY_DID_LEN: usize = _POLY_DID_PREFIX_LEN + _UUID_LEN;
 ///  - "did:poly:1"
 ///  - "DID:poly:..."
 #[derive(Encode, Decode, Default, PartialOrd, Ord, PartialEq, Eq, Clone, Copy, Debug)]
-pub struct IdentityId(u128);
-
-impl IdentityId {
-    /// Generate a randomized `IdentityId`.
-    /// # TODO
-    /// It is not random yet. The implementation could use hash of accountId + nonce.
-    pub fn generate() -> Self {
-        // let v = rand::random::<u128>();
-        IdentityId(0u128)
-    }
-}
+pub struct IdentityId([u8; UUID_LEN]);
 
 impl Display for IdentityId {
     fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
-        write!(f, "did:poly:{:032x}", self.0)
+        write!(f, "did:poly:{:?}", self.0)
     }
 }
 
 impl From<u128> for IdentityId {
     fn from(id: u128) -> Self {
-        IdentityId(id)
+        let mut encoded_id = id.encode();
```

Comment:
We could replace `encode` by `encode_to` and avoid one dynamic allocation and one copy operation.
Something like (it is just an idea, I didn't check it):
```
let mut did = [0; UUID_LEN];
id.encode_to( &did);
IdentityId (did)
```

Reply:
`encode_to` appends the slice to the destination rather than replacing it. We are defining did as fixed-length so we can't append anything to it. Basically, `encode_to` works for vectors but not fixed-size arrays.